File: src/main/java/io/pageobject/generator/locator/SingleElementLocatorPartGenerator.java
package io.pageobject.generator.locator;

import io.pageobject.generator.GeneratorContext;
import org.jsoup.nodes.Element;

public class SingleElementLocatorPartGenerator extends AbstractLocatorPartGenerator {

    public SingleElementLocatorPartGenerator(Element element, LocatorSources... locatorCandidates) {
        super(element, locatorCandidates);
    }

    @Override
    public LocatorPart onlyPart(GeneratorContext context) {
        return generateLastPart(context, ELEMENT_FINDER, 0);
    }

    @Override
    public LocatorPart lastPart(GeneratorContext context, int index) {
        return generateLastPart(context, ELEMENT_FINDER, index);
    }

}

File: lib/locator/single-element-locator-part-generator.js
"use strict";

const AbstractLocatorPartGenerator = require('./abstract-locator-part-generator');

class SingleElementLocatorPartGenerator extends AbstractLocatorPartGenerator {

  constructor(element, ...locatorCandidates) {
    super(element, locatorCandidates);
  }

  onlyPart(context) {
    return super.generateLastPart(context, super.elementFinder, 0);
  }

  lastPart(context, index) {
    return super.generateLastPart(context, super.elementFinder, index);
  }

}

module.exports = SingleElementLocatorPartGenerator;